godef.go

```diff
@@ -225,7 +227,23 @@ func (o orderedObjects) Swap(i, j int)      { o[i], o[j] = o[j], o[i] }
 func done(obj *ast.Object, typ types.Type) {
 	defer os.Exit(0)
 	pos := types.FileSet.Position(types.DeclPos(obj))
-	fmt.Printf("%v\n", pos)
+	if *jsonFlag {
+		jsonMap := make(map[string]string)
+		if pos.Filename != "" {
+			jsonMap["filename"] = pos.Filename
+			if pos.IsValid() {
+				jsonMap["line"] = strconv.Itoa(pos.Line)
+				jsonMap["column"] = strconv.Itoa(pos.Column)
+			}
+		}
+		jsonStr, err := json.Marshal(jsonMap)
```

Comment:
What about:

     data, err := json.Marshal(pos)
     if err != nil {
          fail("cannot marshal JSON output: %v", err)
     }
     fmt.Printf("%s\n", data)

i.e. just marshal the Position itself rather than creating an
intermediary value.

?